### PR TITLE
HDDS-11174. [hsync] Change XceiverClientRatis.watchForCommit to async.

### PR DESCRIPTION
**Pull request by szetszwo**
## What changes were proposed in this pull request?

In the code below, it should just return the replyFuture instead of calling CompletableFuture.get(). This change is useful for HDDS-9844.
```java
//XceiverClientRatis.watchForCommit
      CompletableFuture<RaftClientReply> replyFuture = getClient().async().watch(index, watchType);
      final RaftClientReply reply = replyFuture.get();
```

## What is the link to the Apache JIRA

HDDS-11174

## How was this patch tested?

Updating existing tests.

### CODE REVIEW

**Comment by szetszwo** [comment]
@duongkame and @smengcl , I will keep `RatisBlockOutputStream` unchanged here.  Filed HDDS-11208 to do the further change.  Please review this when you have time.  Thanks.

**Comment by jojochuang** [comment]
This PR won't make it into the HDDS-7593 feature branch before the branch gets merged next week.
Please rebase onto master branch after the branch merge. Thanks!

**Review by jojochuang — COMMENTED** [review]
LGTM. But the PR has a code conflict, and we're about to merge the feature branch into master, so please rebase against master after the branch is merged into master (should take at most another day).

Thanks!

**Comment by szetszwo** [comment]
Sure, will fix the conflicts after the merge.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientSpi.java [hunk 1]
```diff
@@ -178,9 +178,9 @@ public static IOException getIOExceptionForSendCommand(
    * @return reply containing the min commit index replicated to all or majority
    *         servers in case of a failure
    */
-  public abstract XceiverClientReply watchForCommit(long index)
-      throws InterruptedException, ExecutionException, TimeoutException,
-      IOException;
+  public CompletableFuture<XceiverClientReply> watchForCommit(long index) {
```
Comment (smengcl) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientSpi.java:181]: Shall we also change this one to `watchForCommitAsync` without breaking compatibility?

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java [hunk 2]
```diff
@@ -295,46 +294,39 @@ private XceiverClientReply newWatchReply(
   }
 
   @Override
-  public XceiverClientReply watchForCommit(long index)
-      throws InterruptedException, ExecutionException, TimeoutException,
-      IOException {
+  public CompletableFuture<XceiverClientReply> watchForCommit(long index) {
```
Comment (smengcl) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java:297]: Shall we change this one to `watchForCommitAsync` without breaking compatibility?

**Review by smengcl — APPROVED** [review]
Thanks @szetszwo . +1 other than the nits above.

**Comment by smengcl** [comment]
Would you like to take a glance at this @duongkame ?

**Comment by jojochuang** [comment]
Merged. Thanks @smengcl @szetszwo

**Comment by szetszwo** [comment]
@jojochuang , @smengcl , thanks for your helps!

**Pull request merged**